src/main/java/redis/clients/jedis/BinaryShardedJedis.java

```diff
@@ -41,14 +45,19 @@ public void disconnect() {
     for (Jedis jedis : getAllShards()) {
       if (jedis.isConnected()) {
         try {
-          jedis.quit();
+          // need a proper test, probably with mock
+          if (!jedis.isBroken()) {
+            jedis.quit();
+          }
         } catch (JedisConnectionException e) {
           // ignore the exception node, so that all other normal nodes can release all connections.
+          logger.error("Error while QUIT", e);
```

Comment:
Can we reduce this to WARN or INFO, lots of people will have alarms on ERRORs, so if we start adding these it might be tempting for people to turn off the logger completely and they might lose some good information here (Applies in a few other places too)